core/src/main/java/jenkins/security/AgentSecretAction.java

```diff
@@ -0,0 +1,72 @@
+package jenkins.security;
+
+import hudson.model.Action;
+import hudson.model.Computer;
+import hudson.slaves.JNLPLauncher;
+import hudson.slaves.SlaveComputer;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.model.Jenkins;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest2;
+import org.kohsuke.stapler.StaplerResponse2;
+
+public class AgentSecretAction implements Action {
+    private final SlaveComputer computer;
+
+    public AgentSecretAction(SlaveComputer computer) {
+        this.computer = computer;
+    }
+
+    private static final Logger LOGGER = Logger.getLogger(AgentSecretAction.class.getName());
+
+    @Override
+    public String getUrlName() {
+        return "agent-secret";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    public void doGet(StaplerRequest2 req, StaplerResponse2 rsp, @QueryParameter String nodeName) throws IOException {
```

Comment:
Proposing to rename this to `doIndex` 
with the name `doIndex` you could query the secret with `<jenkins>/computer/<computer-name>/agent-secret/` with `doGet` you would need `<jenkins>/computer/<computer-name>/agent-secret/get`